src/cascadia/TerminalSettingsModel/ApplicationState.h

```diff
@@ -16,21 +16,30 @@ Module Name:
 #include "WindowLayout.g.h"
 
 #include <inc/cppwinrt_utils.h>
-#include <til/mutex.h>
-#include <til/throttled_func.h>
-#include "FileUtils.h"
 #include <JsonUtils.h>
 
-// This macro generates all getters and setters for ApplicationState.
-// It provides X with the following arguments:
-//   (type, function name, JSON key, ...variadic construction arguments)
 namespace winrt::Microsoft::Terminal::Settings::Model::implementation
 {
-#define MTSM_APPLICATION_STATE_FIELDS(X)                                                                                \
-    X(std::unordered_set<winrt::guid>, GeneratedProfiles, "generatedProfiles")                                          \
-    X(Windows::Foundation::Collections::IVector<Model::WindowLayout>, PersistedWindowLayouts, "persistedWindowLayouts") \
-    X(Windows::Foundation::Collections::IVector<hstring>, RecentCommands, "recentCommands")                             \
-    X(Windows::Foundation::Collections::IVector<winrt::Microsoft::Terminal::Settings::Model::InfoBarMessage>, DismissedMessages, "dismissedMessages")
+    // If a property is Shared, then it'll be stored in `state.json`, and used
+    // in both elevated and unelevated instances of the Terminal. If a property
+    // is marked Local, then it will have separate values for elevated and
+    // unelevated instances.
+    enum FileSource : int
+    {
+        Shared = 0x1,
+        Local = 0x2
+    };
+    DEFINE_ENUM_FLAG_OPERATORS(FileSource);

```

Comment:
I believe you can save yourself quite some work by splitting the `MTSM_APPLICATION_STATE_FIELDS` macro into two. It would allow you to drop the entire `FileSource` logic alltogether.

Reply:
Any interest in doing this? @lhecker this something you would require?

Reply:
Unless I block a PR, I wouldn't consider my feedback is obligatory. 😄
I just believe that this might simplify some of the code quite a bit, since it would replace ifs and elses with... well nothing basically. That's why I think it's not a bad idea to split the macro.